remsfal-core/src/main/java/de/remsfal/core/api/project/ChatEndpoint.java

```diff
@@ -0,0 +1,294 @@
+
+    @DELETE
+    @Path("/{sessionId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Delete chat session")
+    @APIResponse(responseCode = "200", description = "Chat session deleted")
+    @APIResponse(responseCode = "400", description = "Invalid input")
+    @APIResponse(responseCode = "404", description = "Project, task, or chat session not found")
+    @APIResponse(responseCode = "500", description = "Internal server error")
+    @APIResponse(responseCode = "401", description = "No user authentication provided via session cookie")
+    Response deleteChatSession(
+            @PathParam("sessionId") @NotNull @UUID String sessionId);
+
+    @PUT
+    @Path("/{sessionId}/status")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Update chat session status")
+    @APIResponse(responseCode = "200", description = "Chat session status updated")
+    @APIResponse(responseCode = "400", description = "Invalid input")
+    @APIResponse(responseCode = "404", description = "Project, task, or chat session not found")
+    @APIResponse(responseCode = "500", description = "Internal server error")
+    @APIResponse(responseCode = "401", description = "No user authentication provided via session cookie")
+    Response updateChatSessionStatus(
+            @PathParam("sessionId") @NotNull @UUID String sessionId,
+            @Parameter(description = "New status for the chat session", required = true)
+            @Valid @NotNull ChatSessionModel.Status status);
+
+    @POST
+    @Path("/{sessionId}/join")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Join chat session as a participant - observer")
+    @APIResponse(responseCode = "200", description = "Chat session joined")
+    @APIResponse(responseCode = "400", description = "Invalid input")
+    @APIResponse(responseCode = "404", description = "Project, task, or chat session not found")
+    @APIResponse(responseCode = "500", description = "Internal server error")
+    @APIResponse(responseCode = "401", description = "No user authentication provided via session cookie")
+    Response joinChatSession(
+            @PathParam("sessionId") @NotNull @UUID String sessionId);
+
+    @POST
+    @Path("/{sessionId}/handle")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Join chat session as a participant - handler")
+    @APIResponse(responseCode = "200", description = "Chat session joined")
+    @APIResponse(responseCode = "400", description = "Invalid input")
+    @APIResponse(responseCode = "404", description = "Project, task, or chat session not found")
+    @APIResponse(responseCode = "500", description = "Internal server error")
+    @APIResponse(responseCode = "401", description = "No user authentication provided via session cookie")
+    Response handleChatSession(
```

Comment:
Same security question?

Reply:
explained on this [response](https://github.com/remsfal/remsfal-backend/pull/239#discussion_r1830916029)